pkg/reconcilers/service.go

```diff
@@ -8,6 +8,32 @@ import (
 	v1 "k8s.io/api/core/v1"
 )
 
+func ServiceMutator(opts ...MutateFn) MutateFn {
+	return func(existingObj, desiredObj common.KubernetesObject) (bool, error) {
```

Comment:
Q
Why the need for these? Is this something related to move to deployments?

Reply:
Yep this is related to move to Deployments. See this [comment](https://github.com/3scale/3scale-operator/pull/929#discussion_r1502910357) for more context.